server/src/main/java/org/elasticsearch/index/engine/TranslogOperationAsserter.java

```diff
@@ -43,11 +46,32 @@ public boolean assertSameIndexOperation(Translog.Index o1, Translog.Index o2) th
                     return super.assertSameIndexOperation(synthesizeSource(engineConfig, o1), o2)
                         || super.assertSameIndexOperation(o1, synthesizeSource(engineConfig, o2));
                 }
+                // TODO: Investigate why we still have different precision even after synthesizing the source, and remove this check
```

Comment:
Can we call this a selective test-mute in this comment - and reference the CI issue?